ci/util/fbuild_runner.py

```
@@ -75,6 +111,81 @@ def _get_output(quiet: bool, log_file: IO[str] | None) -> IO[str]:
+
+
+@lru_cache(maxsize=1)
+def fbuild_supports_compile_many() -> bool:
+    """Return whether the active fbuild executable exposes ``compile-many``."""
+    import subprocess
+
+    fbuild_exe = get_fbuild_executable()
+    if fbuild_exe is None:
+        return False
+    try:
+        proc = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)